src/deviation-check/deviation-check.ts

```diff
@@ -1,14 +1,16 @@
-import { HUNDRED_PERCENT } from '../constants';
+import { HUNDRED_PERCENT, UINT256_MAX } from '../constants';
 import { logger } from '../logger';
 import { abs } from '../utils';
 
-export const calculateDeviationPercentage = (initialValue: bigint, updatedValue: bigint) => {
-  const delta = updatedValue - initialValue;
-  const absoluteDelta = abs(delta);
-
-  // Avoid division by 0
-  const absoluteInitialValue = initialValue === 0n ? 1n : abs(initialValue);
+export const calculateDeviationPercentage = (
+  initialValue: bigint,
+  updatedValue: bigint,
+  deviationReference: bigint
+) => {
+  const absoluteDelta = abs(updatedValue - initialValue);
```

Comment:
After this line, we need to check that `absoluteDelta` is zero. If it is, we must return 0 deviation in percentage like in the [contract](https://github.com/api3dao/airnode-protocol-v1/blob/main/contracts/api3-server-v1/extensions/BeaconSetUpdatesWithPsp.sol#L162). One can say that if `absoluteDelta` is zero, it will return zero at the end of the function, however if  `absoluteInitialValue` is also zero, it will return infinite deviation in percentage.

Reply:
Right 🤦